charts/prometheus-node-exporter/templates/daemonset.yaml

```diff
@@ -39,6 +39,7 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       serviceAccountName: {{ include "prometheus-node-exporter.serviceAccountName" . }}
+      {{- with .Values.nodeselector }}
```

Comment:
@marciogmorales 
You miss an `{{- end }}` in line 278 if you'd like to keep `{{- with .Values.nodeselector }}`.